producer/datapath.go

```diff
@@ -109,34 +114,40 @@ func SendPFCPRules(smContext *smf_context.SMContext) {
 	for ip, pfcp := range pfcpPool {
 		sessionContext, exist := smContext.PFCPContext[ip]
 		if !exist || sessionContext.RemoteSEID == 0 {
-			pfcp_message.SendPfcpSessionEstablishmentRequest(
+			err := message.SendPfcpSessionEstablishmentRequest(
 				pfcp.nodeID, smContext, pfcp.pdrList, pfcp.farList, nil, pfcp.qerList, pfcp.port)
+			if err != nil {
+				logger.PduSessLog.Errorf("send pfcp session establishment request failed: %v for UPF[%v, %v]: ", err, pfcp.nodeID, pfcp.nodeID.ResolveNodeIdToIp())
+			}
 		} else {
-			pfcp_message.SendPfcpSessionModificationRequest(
+			err := message.SendPfcpSessionModificationRequest(
 				pfcp.nodeID, smContext, pfcp.pdrList, pfcp.farList, nil, pfcp.qerList, pfcp.port)
+			if err != nil {
+				logger.PduSessLog.Errorf("send pfcp session modification request failed: %v for UPF[%v, %v]: ", err, pfcp.nodeID, pfcp.nodeID.ResolveNodeIdToIp())
+			}
 		}
 	}
 }
 
-func removeDataPath(smContext *smf_context.SMContext, datapath *smf_context.DataPath) {
+func removeDataPath(smContext *context.SMContext, datapath *context.DataPath) {
```

Comment:
FYI, `smContext` seem to be unused. Do we need to keep it?

Reply:
Removed